File: python/JogoDeAdivinhacao.py
import os
import random
import time

def banner():

    os.system('cls')

    print("=======================")
    print("= Jogo De Adivinhação =")
    print("=======================")

def facil():

    banner()

    print("\nA dificuldade escolhida foi: Fácil - 1 a 10")

    valorSecreto = random.randint(1, 10)

    while True:
        palpite = int(input("Digite seu palpite: "))
        if palpite == valorSecreto:
            print ("Parabéns! Você acertou! O valor secreto era "+str(valorSecreto)+"!!!")
            jogarDenovo = str(input("Deseja jogar novamente na mesma dificuldade (s/n)? "))
            if jogarDenovo == 's':
                facil()
            elif jogarDenovo == "n":
                banner()
                print("\nVoltando para a tela inicial!")
                time.sleep(3.5)
                inicio()
            break
        elif palpite > valorSecreto:
            print("É menor\n")
        elif palpite < valorSecreto:
            print("É maior\n")

def medio():

    banner()

    print("\nA dificuldade escolhida foi: Médio - 1 a 50\n")

    valorSecreto = random.randint(1, 50)

    while True:
        palpite = int(input("Digite seu palpite: "))
        if palpite == valorSecreto:
            print("Parabéns! Você acertou! O valor secreto era "+str(valorSecreto)+"!!!")
            jogarDenovo = str(input("Deseja jogar novamente na mesma dificuldade (s/n)? "))
            if jogarDenovo == "s":
                medio()
            elif jogarDenovo == "n":
                banner()
                print("\nVoltando para a tela inicial!")
                time.sleep(3.5)
                inicio()
                break
        elif palpite > valorSecreto:
            print("É menor\n")
        elif palpite < valorSecreto:
            print("É maior\n")

def dificil():

    banner()

    print("\nA difiuldade escolhida foi: Difícil - 1 a 100\n")

    valorSecreto = random.randint(1, 100)

    while True:
        palpite = int(input("Digite seu palpite: "))
        if palpite == valorSecreto:
            print("Parabéns! Você acertou! O valor secreto era "+str(valorSecreto)+"!!!")
            jogarDenovo = str(input("Deseja jogar novamente na mesma dificuldade (s/n)? "))
            if jogarDenovo == "s":
                dificil()
            elif jogarDenovo == "n":
                banner()
                print("\nVoltando para a tela inicial!")
                time.sleep(3.5)
                inicio()
                break
        elif palpite > valorSecreto:
            print("É menor\n")
        elif palpite < valorSecreto:
            print("É maior\n")

def inicio():

    banner()

    print("Bem vindo(a) ao Jogo de Adivinhação!\n")
    print("   TABELA DE DIFICULDADES")
    print("  ========================")
    print(" || 1. Fácil - 1 a 10    ||")
    print(" || 2. Médio - 1 a 50    ||")
    print(" || 3. Difícil - 1 a 100 ||")
    print("  ========================")

    dificuldade = str(input("Escolha a dificuldade do jogo: "))

    if dificuldade == "1":
        facil()
    elif dificuldade == "2":
        medio()
    elif dificuldade == "3":
        dificil()
    else:
        banner()
        print("\nEssa dificuldade não está disponível!\n")
        print("Voltando para a tela de início!!!")
        time.sleep(3.5)
        inicio()
        
inicio()

File: C/JogoDeAdivinhacao.c
#include <stdio.h>
#include <stdlib.h>
#include <windows.h>
#include <time.h>
#include <stdbool.h>

void banner(){
    system("cls");
    printf(" =======================\n");
    printf("|| Jogo De Adivinhacao ||\n");
    printf(" =======================\n");
}

void facil(){
    int valorSecreto, palpite;
    char jogarDenovo;

    banner();
    printf("\nA dificuldade escolhida foi: Facil - 1 a 10\n\n");

    valorSecreto = rand() % 10 + 1;

    while (true) {
        printf("Digite seu palpite: ");
        scanf("%d", &palpite);

        if (palpite == valorSecreto) {
            printf("\nParabens! Voce acertou! O valor secreto era %d!!!\n", valorSecreto);
            printf("Deseja jogar novamente na mesma dificuldade? (s/n): ");
            scanf(" %c", &jogarDenovo);

            if (jogarDenovo == 's') {
                facil();
                return;
            } else if (jogarDenovo == 'n') {
                banner();
                printf("\nVoltando para a tela inicial...\n");
                Sleep(3500);
                return;
            }
        } else if (palpite > valorSecreto) {
            printf("\nE menor!\n");
        } else if (palpite < valorSecreto) {
            printf("\nE maior!\n");
        }
    }
}

void medio(){
    int valorSecreto, palpite;
    char jogarDenovo;

    banner();
    printf("\nA dificuldade escolhida foi: Medio - 1 a 50\n\n");

    valorSecreto = rand() % 50 + 1;

    while (true) {
        printf("Digite seu palpite: ");
        scanf("%d", &palpite);

        if (palpite == valorSecreto) {
            printf("\nParabens! Voce acertou! O valor secreto era %d!!!\n", valorSecreto);
            printf("Deseja jogar novamente na mesma dificuldade? (s/n): ");
            scanf(" %c", &jogarDenovo);

            if (jogarDenovo == 's') {
                medio();
                return;
            } else if (jogarDenovo == 'n') {
                banner();
                printf("\nVoltando para a tela inicial...\n");
                Sleep(3500);
                return;
            }
        } else if (palpite > valorSecreto) {
            printf("\nE menor!\n");
        } else if (palpite < valorSecreto) {
            printf("\nE maior!\n");
        }
    }
}

void dificil(){
    int valorSecreto, palpite;
    char jogarDenovo;

    banner();

    printf("\nA dificuldade escolhida foi: Dificil - 1 a 100\n\n");

    valorSecreto = rand() % 50 + 1;

    while (true) {
        printf("Digite seu palpite: ");
        scanf("%d", &palpite);

        if (palpite == valorSecreto) {
            printf("\nParabens! Voce acertou! O valor secreto era %d!!!\n", valorSecreto);
            printf("Deseja jogar novamente na mesma dificuldade? (s/n): ");
            scanf(" %c", &jogarDenovo);

             if (jogarDenovo == 's') {
                dificil();
                return;
            } else if (jogarDenovo == 'n') {
                banner();
                printf("\nVoltando para a tela inicial...\n");
                Sleep(3500);
                return;
            }
        } else if (palpite > valorSecreto) {
            printf("\nE menor!\n");
        } else if (palpite < valorSecreto) {
            printf("\nE maior!\n");
        }
    }
}

int main(){
    srand(time(NULL));

    int dificuldade;

    while (true) {
        banner();
        printf("\nBem vindo(a) ao Jogo de Adivinhacao!\n");
        printf("\n   TABELA DE DIFICULDADES\n");
        printf("  ========================\n");
        printf(" || 1. Facil   - 1 a 10  ||\n");
        printf(" || 2. Medio   - 1 a 50  ||\n");
        printf(" || 3. Dificil - 1 a 100 ||\n");
        printf("  ========================\n\n");

        printf("Escolha a dificuldade: ");
        scanf("%d", &dificuldade);

        if (dificuldade == 1) {
            facil();
        } else if (dificuldade == 2) {
            medio();
        } else if (dificuldade == 3) {
            dificil();
        } else {
            printf("\nOpcao invalida! Tente novamente.\n");
            Sleep(2000);
        }
    }

    return 0;
}
